fairseq/models/wav2vec/wav2vec2.py

```diff
@@ -454,21 +473,23 @@ def sample_negatives(self, y, num):
         y = y.view(-1, fsz)  # BTC => (BxT)C
 
         cross_high = tsz * bsz
-        high = tsz
+        high = num
```

Comment:
num = number of actual timesteps to sample for.
tsz = number of timesteps in the batch

in your case num should equal tsz (same as negatives from everywhere + sampling extra negs that wont get used) which would already be correct since you dont shrink y. so you dont need to change this.